### PR TITLE
Add "No Bare Raises" checker

### PR DESCRIPTION
**Pull request by d1618033**
The 'No Bare Raises' checker is inspired by [this issue](https://github.com/pandas-dev/pandas/issues/30999)

A lot of the changes are whitespace (end of line character), so it's easier to select ignore whitespace in the diff

### CODE REVIEW

**Comment by d1618033** [comment]
@m-burst Should I close this?

**Comment by m-burst** [comment]
Hi @d1618033,

Sorry I didn't have time to look through the changes earlier.

Personally I'm not sure that this rule should be introduced for all exceptions.
In many cases the `pytest.raises` check will contain a more specific exception than `ValueError` so the message check would be extraneous. Moreover, since any further changes to exception message will break the tests, this will cause inconvenience to developers.
A reasonable way to introduce such rule might be to add a configurable list of exception classes and force the `match` parameter only with these classes.

**Comment by d1618033** [comment]
Hi @m-burst,
Good points. Is there an easy to way to add configuration to a flake8 plugin? 
I'm having difficulties doing this in an idiomatic way using flake8-plugin-utils (see https://github.com/afonasev/flake8-plugin-utils/issues/6)

**Comment by m-burst** [comment]
@d1618033, I've started experimenting with adding configuration to the plugin but don't have a working prototype yet

**Comment by d1618033** [comment]
@m-burst Could you please review this again? I made the exceptions for which this rule applies configurable (by default, no exceptions, i.e, the rule is off).

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .PHONY: init test test-cov lint format
 
 CODE = flake8_pytest_style
-TEST = pytest --verbosity=2 --showlocals --strict --cov=$(CODE)
+TEST = poetry run pytest --verbosity=2 --showlocals --strict --cov=$(CODE)
```
Comment (m-burst) [inline Makefile]: Please update to the latest master, this is fixed there.
In future PRs please try not to mix unrelated changes.

File: flake8_pytest_style/errors.py [hunk 2]
```diff
@@ -46,3 +46,8 @@ class PatchWithLambda(Error):
 class UnittestAssertion(Error):
     code = 'PT009'
     message = "use a regular assert instead of unittest-style '{assertion}'"
+
+
+class NoBareRaises(Error):
+    code = 'PT010'
+    message = 'set the match parameter in pytest.raises'
```
Comment (m-burst) [inline flake8_pytest_style/errors.py]: I suggest we add the exception name here, i.e.:
`message = 'set the match parameter in pytest.raises({exception})'`

File: flake8_pytest_style/plugin.py [hunk 3]
```diff
@@ -17,10 +17,14 @@ class PytestStylePlugin(Plugin[Config]):
 
     @classmethod
     def add_options(cls, option_manager: OptionManager) -> None:  # pragma: no cover
-        pass
+        option_manager.add_option(
+            '--no-bare-raises-exceptions',
```
Comment (m-burst) [inline flake8_pytest_style/plugin.py]: I guess we should prefix all our config option names with `pytest-`

File: flake8_pytest_style/visitors.py [hunk 4]
```diff
@@ -164,6 +168,22 @@ def _check_parametrize_call(self, node: ast.Call) -> None:
 
         self._check_parametrize_values(node, args.values, multiple_names)
 
+    def _check_raises_call(self, node: ast.Call) -> None:
+        """Checks for all violations regarding `pytest.raises` calls."""
+        args = get_simple_call_args(node)
+        if not args:
+            return
+        exception = args.get_argument('expected_exception', position=0)
+        if not isinstance(exception, ast.Name):
```
Comment (m-burst) [inline flake8_pytest_style/visitors.py]: We should use `get_qualname` to handle cases where the passed argument is a dotted name, e.g. `socket.error` (and also add a test for that)

File: README.md [hunk 5]
```diff
@@ -38,6 +38,21 @@ and `mocker.patch('target', lambda *args: 7)` is an error
 
 * `PT009 use a regular assert instead of unittest-style '{assertion}'`
 
+* `PT010 set the match parameter in pytest.raises`  
+e.g. `pytest.raises(ValueError, match='exception text')` is OK,
+and `pytest.raises(ValueError)` is an error.
+    * Since this is a fairly disruptive rule, it is by default off. 
+      You can turn it on by specifying in your `setup.cfg` file
```
Comment (m-burst) [inline README.md]: `setup.cfg` is not the only way to configure Flake8, so I would phrase this differently:
```suggestion
      You can turn it on by specifying in your configuration (e.g. `setup.cfg`) 
```

Comment (m-burst) [inline README.md]: I suggest to organize the part of README regarding configuration in a different way. We can add a separate section with the list of all the options with their descriptions, and simply say here that the list of exceptions is controlled by `no-bare-raises-exceptions` option.

**Comment by d1618033** [comment]
Thank you so much for your review! I will work on making the changes you requested.

> I also think we should add another rule before we add this one: to disallow pytest.raises() without specifying an exception.

I think this is an excellent idea! 

> Would you mind if I did the necessary changes in the PR myself?

Sure.

**Comment by m-burst** [comment]
Hi @d1618033,
Thanks for your work!
I've completed this and released v0.3.0.

**Pull request merged**